File: lib/features/buyer/utils/search_trends.dart
List<String> resolveTrendingSearchTerms({
  required List<String> configuredTerms,
  required List<String> fallbackTerms,
  int limit = 8,
}) {
  final source = configuredTerms.any((term) => term.trim().isNotEmpty)
      ? configuredTerms
      : fallbackTerms;
  final seen = <String>{};
  final resolved = <String>[];

  for (final term in source) {
    final clean = term.trim();
    if (clean.isEmpty) continue;
    final key = clean.toLowerCase();
    if (seen.contains(key)) continue;
    seen.add(key);
    resolved.add(clean);
    if (resolved.length >= limit) break;
  }

  return resolved;
}

File: landing/src/lib/marketplace/search-trends.ts
export function resolveTrendingSearchTerms({
  configuredTerms,
  fallbackTerms,
  limit = 8,
}: {
  configuredTerms: string[];
  fallbackTerms: string[];
  limit?: number;
}) {
  const source = configuredTerms.some((term) => term.trim().length > 0)
    ? configuredTerms
    : fallbackTerms;
  const seen = new Set<string>();
  const resolved: string[] = [];

  for (const term of source) {
    const clean = term.trim();
    const key = clean.toLowerCase();
    if (!clean || seen.has(key)) {
      continue;
    }
    seen.add(key);
    resolved.push(clean);
    if (resolved.length >= limit) {
      break;
    }
  }

  return resolved;
}
